framework/src/kernels/ADScalarKernel.C

```diff
@@ -0,0 +1,82 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADScalarKernel.h"
+
+#include "Assembly.h"
+
+InputParameters
+ADScalarKernel::validParams()
+{
+  InputParameters params = ScalarKernelBase::validParams();
+  return params;
+}
+
+ADScalarKernel::ADScalarKernel(const InputParameters & parameters)
+  : ScalarKernelBase(parameters), _u(_var.adSln()), _jacobian_already_computed(false)
+{
+#ifndef MOOSE_GLOBAL_AD_INDEXING
+  mooseError("ADScalarKernel can only be used with global AD indexing.");
+#endif
+}
+
+void
+ADScalarKernel::reinit()
+{
+  _jacobian_already_computed = false;
+}
+
+void
+ADScalarKernel::computeResidual()
+{
+  prepareVectorTag(_assembly, _var.number());
+
+  for (_i = 0; _i < _var.order(); _i++)
+    _local_re(_i) += raw_value(computeQpResidual());
```

Comment:
If this compiles, we must have a `using MetaPhysicL::raw_value` somewhere 🤔

Reply:
It doesn't occur in the framework anywhere, and I don't see any other `raw_value` method defined in MOOSE, so I'm guessing this was done at the libMesh level somewhere.

Reply:
Ah this has to be argument-dependent lookup